ably/realtime_conn.go

```diff
@@ -115,7 +115,7 @@ func (c *Connection) dial(proto string, u *url.URL) (conn proto.Conn, err error)
 	query := u.Query()
 	query.Add("heartbeats", "true")
 	u.RawQuery = query.Encode()
-	timeout := c.opts.httpOpenTimeout()
+	timeout := c.opts.realtimeRequestTimeout()
```

Comment:
@tcard , I have reverted this change that belong to https://github.com/ably/ably-go/pull/312, because all of the channel
attach tests suddenly started breaking, because default `httpOpenTimeout` doesn't seem to be enough to get state from
`conneting` to `connected`. Most of the transitioner based tests just failed with error saying, went from `connecting` to `disconnected` cuz of short timeout

Reply:
Opened #325 to deal with this separately.